classes/suggested-tasks/local-tasks/providers/class-debug-display.php

```diff
@@ -27,12 +27,17 @@ class Debug_Display extends Local_OneTime_Tasks_Abstract {
 	const ID = 'wp-debug-display';
 
 	/**
-	 * Check if the task condition is met.
+	 * Check if the task condition is satisfied.
+	 * (bool) true means that the task condition is satisfied, meaning that we don't need to add the task or task was completed.
 	 *
 	 * @return bool
 	 */
 	public function check_task_condition() {
-		return ( ! defined( 'WP_DEBUG_DISPLAY' ) || ! WP_DEBUG_DISPLAY ) ? true : false;
+		/**
+		 * For WP_DEBUG_DISPLAY to do anything, WP_DEBUG must be enabled (true).
+		 * link: https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/#wp_debug_display
+		 */
+		return defined( 'WP_DEBUG' ) && ( false === WP_DEBUG || ( WP_DEBUG && ( ! defined( 'WP_DEBUG_DISPLAY' ) || ! WP_DEBUG_DISPLAY ) ) ) ? true : false; // @phpstan-ignore-line
```

Comment:
Not a blocker, just a suggestion, minor readability tweak:
* Splits long condition to multiple lines
* Removes one of the nested `WP_DEBUG` checks
* Removes the `? true : false;` part since the condition already returns a bool result
```suggestion
		return (
			defined( 'WP_DEBUG' ) && (
				false === WP_DEBUG ||
				( ! defined( 'WP_DEBUG_DISPLAY' ) || ! WP_DEBUG_DISPLAY )
			)
		);
```

Reply:
Sure!

I usually leave the `? true : false;` part for readability and phpstan complained about doing `Removes one of the nested WP_DEBUG checks` (which I also left for readability since it is literal "translation" of what is said in the WP docs), but we can change it if is clear enough what is done.